### DISCUSSION TITLE
How to pass arguments to npm package scripts?

### DISCUSSION THREAD

**Discussion by michael-land** [pnpm/pnpm#4844] [Q&A] [answered]
I'm new to pnpm.

I have the following npm scripts with yarn

```
  "scripts": {
    "dev": "turbo run dev --filter=${0}..."
  }

```

Whenever I do `yarn dev program_name` it will replace `${0}` with `program_name`,  become `turbo run dev --filter=program_name...`

What's the equivalent way to do the same in pnpm?

**Comment by zkochan** [comment, marked as answer]
I don't think this is currently supported in pnpm.

**Reply by michael-land** [reply]
ah, thanks

**Comment by ronaldruzicka** [comment]
Is it still the case? We would really need this in our project. We wanted to use it via `sh -c` which then allows to add arguments, but it doesn't work on Windows, which some people use 😕

**Comment by iobuhov** [comment]
What I do personally, I just pass this arguments to my script:
So, in my package.json
```
"scripts": {
  "build": "turbo run build"
}
```

Then if you run `pnpm run build --filter=my-package` it will become `turbo run build --filter=my-package` pnpm just pass arguments to the script.

Logs from our GH action
```
Run pnpm run build --concurrency=5 --filter "...[origin/main]"
  pnpm run build --concurrency=5 --filter "...[origin/main]"
  shell: /usr/bin/bash -e {0}
  env:
    since_flag: --filter "...[origin/main]"
    PNPM_HOME: /home/runner/setup-pnpm/node_modules/.bin
    main_sha: 2aa431779408f3a6d4604bac5b17fd424a418c43
    MSYS: winsymlinks:nativestrict
    task_concurrency: 5
    NODE_OPTIONS: --max-old-space-size=5120 --max_old_space_size=5120

> web-widgets@ build /home/runner/work/web-widgets/web-widgets
> turbo run build "--concurrency=5" "--filter" "...[origin/main]"
```

**Reply by ronaldruzicka** [reply]
Yes I know it can pass arguments, but there is no way how to reference them in `package.json`. I basically wanted to shorten the scripts as much as possible. So we ended up using `sh -c`
```json
"dev": "sh -c 'turbo run dev --filter=@abis/${0}'",
```
so all I have to write is
```bash
pnpm dev my-app
```
Just Windows users will have to use Git Bash or some other terminal that can execute bash scripts. But since `yarn` and `npm` have this functionality I really thought that `pnpm` has it as well

**Comment by neutron0831** [comment]
@michael-land
I was looking for the same thing and find a solution which uses env variable[^1]. I hope this helps you.

[^1]: https://pnpm.io/pnpm-cli#differences-vs-npm

```json
  "scripts": {
    "dev": "turbo run dev --filter=@abis/$program_name"
  }
```

```sh
program_name=my-app pnpm dev
```
